application/src/main/java/org/opentripplanner/service/paging/PagingService.java

```diff
@@ -184,12 +186,33 @@ private PageCursorFactory mapIntoPageCursorFactory(@Nullable PageType currentPag
       searchWindowUsed
     );
 
+    handleFirstSearch(currentPageType, factory);
+
     if (pageCursorInput != null) {
       factory = factory.withRemovedItineraries(pageCursorInput);
     }
     return factory;
   }
 
+  /**
+   * If the first search is an arrive by search, add the latest itinerary's start time
+   * to allow setting cursor information correctly for the page cursor of the next page.
+   */
+  private void handleFirstSearch(@Nullable PageType currentPageType, PageCursorFactory factory) {
```

Comment:
Consider: I think this method is an abstraction leak. All business logic to calculate the `PageCursor` should be in the `PageCursorFactory` the Service should not know about the details. To fix this, all you need to do is:
 -  make the `resolvePageTypeForTheFirstSearch` private
 - move this logic into `PageCursorFactory`
 - pass in `lastItineraryDeparture` to the factory. 
 
 Sorry for not pointing out this before - I only had a hunch there was something wrong here, and only after a break did it appear to me what was wrong 🤔.

Reply:
I moved the logic into `PageCursorFactory`, I passed the itinerary list as a parameter to the `withOriginalSearch` method instead of `lastItineraryDeparture` because this seemed more logical.

Reply:
I changed the `PageCursorFactory` to take an instant instead of the list of itineraries as we discussed in the dev meeting